src/main/java/de/tum/in/www1/exerciseapp/domain/enumeration/ParticipationState.java

```diff
@@ -4,7 +4,7 @@
  * The ParticipationState enumeration.
  */
 public enum ParticipationState {
-    UNINITIALIZED(0), REPO_COPIED(1), REPO_CONFIGURED(2), BUILD_PLAN_COPIED(3), BUILD_PLAN_CONFIGURED(4), INITIALIZED(5);
+    UNINITIALIZED(0), REPO_COPIED(1), REPO_CONFIGURED(2), INACTIVE(3), BUILD_PLAN_COPIED(4), BUILD_PLAN_CONFIGURED(5), INITIALIZED(6);
```

Comment:
I think this could lead to problems with the enum, because 3 was "build plan copied" before and now is "inactive". So whenever 3 was used, it will not work. I suggest to add inactive at the end and give it the integer 6

Reply:
No, I believe it's the other way around. Numeric values cannot be used in code outside of enum class definition in Java, unless you have a public getter (smth like public Integer getStateNum() { return stateNumber; }) and do some comparisons on numbers outside of enum class (which is really bad practice, I would guess). We have a hasCompletedState(..) method instead.

On the contrary, the methods that change the participation state (copyBuildPlan(..), configureBuildPlan(..), etc.) check that the participation is in the proper state before changing it.
E.g., copyBuildPlan(..) method checks that the current participation state is lower than BUILD_PLAN_COPIED (using hasCompletedState(..) method). This way, if INACTIVE is 6, it will not pass the check, and the request to resume participation is ignored. Having it in position 3 seems most logical, because an inactive participation has a configured repository, but no build plan.

Reply:
sounds reasonable!